packages/design-preview/src/components/index.tsx

```diff
@@ -1,17 +1,43 @@
+import { useEffect, useState } from '@wordpress/element';
 import Sidebar from './sidebar';
-import type { StyleVariation } from '@automattic/design-picker';
+import SitePreview from './site-preview';
+import type { StyleVariation } from '@automattic/design-picker/src/types';
 import './style.scss';
 
 interface PreviewProps {
+	previewUrl: string;
 	title?: string;
 	description?: string;
 	variations?: StyleVariation[];
 }
 
-const Preview: React.FC< PreviewProps > = ( { title, description, variations = [] } ) => {
+const Preview: React.FC< PreviewProps > = ( {
+	previewUrl,
+	title,
+	description,
+	variations = [],
+} ) => {
+	const [ activeVariation, setActiveVariation ] = useState< StyleVariation | undefined >();
+	const handleVariationClick = ( variation: StyleVariation ) => {
+		setActiveVariation( variation );
+	};
+
+	useEffect( () => {
+		if ( ! activeVariation ) {
+			setActiveVariation( variations[ 0 ] );
+		}
+	}, [ variations, activeVariation ] );
+
 	return (
 		<div className="design-preview">
-			<Sidebar title={ title } description={ description } variations={ variations } />
+			<Sidebar
+				title={ title }
+				description={ description }
+				variations={ variations }
+				activeVariation={ activeVariation }
+				onVariationClick={ handleVariationClick }
```

Comment:
[nit] (not sure if you prefer this?)

```suggestion
				onVariationClick={ setActiveVariation }
```